core/libs/shared/src/staged.rs

```diff
@@ -112,23 +58,9 @@ where
 impl<Base, Staged> StagedTree<Base, Staged>
 where
     Base: TreeLike,
-    Staged: TreeLikeMut<F = Base::F>,
+    Staged: TreeLike<F = Base::F>,
 {
-    pub fn new(base: Base, mut staged: Staged) -> Self {
-        let mut prunable = vec![];
-        for id in staged.ids() {
-            if let Some(staged) = staged.maybe_find(id) {
-                if let Some(base) = base.maybe_find(id) {
-                    if staged == base {
-                        prunable.push(*id);
-                    }
-                }
-            }
-        }
-
-        for id in prunable {
-            staged.remove(id);
-        }
+    pub fn new(base: Base, staged: Staged) -> Self {
         Self { base, staged, removed: HashSet::new() }
     }
 }
```

Comment:
in order to support staging immutable changes, changes are no longer pruned on stage